sql/expression/arithmetic.go

```diff
@@ -134,34 +134,120 @@ func (a *Arithmetic) Type() sql.Type {
 
 	// applies for + and - ops
 	if isInterval(a.LeftChild) || isInterval(a.RightChild) {
-		// TODO: we might need to truncate precision here
-		return types.DatetimeMaxPrecision
+		// TODO: need to use the precision stored in datetimeType; something like
+		// return MustCreateDatetimeType(sqltypes.Datetime, ...)
+		return types.Datetime
```

Comment:
Does this affect any results? Looks like we only use the `Datetime.precision` value when converting to a time type, but I couldn't think of a quick arithmetic op example that would trigger this. I did notice the plan changes for tpcc dropped the precision, so seems like casting those values to times would have different behavior before/after. 

Seems like we're close enough here that it'd be nice to finish this one if we can. Otherwise, it is safer (less of a behavior change) to continue returning `types.DatetimeMaxPrecision`?

Reply:
This somewhat affects results; more specifically it impacts the result type.

When performing arithmetic with DateTime, the precision determines if the result will be a `Decimal` or `Integer`.
Arithmetic involving a DateTime(0) should return an `Integer`.
Otherwise, arithmetic involving `DateTime(x)`, returns a `Decimal(<prec>, x)`.

After reimplementing the `Arithmetic.Type()` and `Div.Type()`, the tests with `STR_TO_DATE` broke because they were returning the wrong type.
We started returning Decimal instead of Integers because of the DateTime Precision.
```golang
{
	Query: "SELECT STR_TO_DATE('01,5,2013 09:30:17','%d,%m,%Y %h:%i:%s') - (STR_TO_DATE('01,5,2013 09:30:17','%d,%m,%Y %h:%i:%s') - INTERVAL 1 SECOND)",
	Expected: []sql.Row{{int64(1)}},
},
```

This also lead to the discovery of this broken query:
```golang
{
	// TODO:  need to properly handle datetime precision
	Query:    `SELECT STR_TO_DATE('01,5,2013 09:30:17','%d,%m,%Y %h:%i:%s %f') - (STR_TO_DATE('01,5,2013 09:30:17','%d,%m,%Y %h:%i:%s') - INTERVAL 1 SECOND)`,
	Expected: []sql.Row{{int64(1)}},
},
```